full-node/sov-stf-runner/src/prover_service/mod.rs

```diff
@@ -19,6 +19,16 @@ pub enum RollupProverConfig {
     Prove,
 }
 
+/// Indicates the status of the DA proof submission.
+pub enum ProofSubmissionStatus {
+    /// Proof was submitted to the DA.
+    Success,
+    /// Proof generation is still in progress.
+    ProvingInProgress,
```

Comment:
Yes, this is a good point. Currently, the caller waits in a loop with some delay until the proof is ready. I think we will need something like a `future` or channel subscription, but I would like to defer the implementation after we have the fork management mechanism from @citizen-stig ready. This is because we will send proofs to DA only when the corresponding block is finalized, so we have to design with the `ForkMenager` in mind.

Reply:
Sounds good! I'd suggest something like a [oneshot](https://docs.rs/tokio/1.34.0/tokio/sync/oneshot/index.html) channel (or some other library that implements a similar functionality) as it does perform well and is quite robust, but you're right that we should prioritize consistency with other parts of the code